### ISSUE TITLE
Order status changes to processing without successful payment

### ISSUE THREAD

**Issue by aasghar551** [wallace-stev/epush-selcom-wp#1]
Hi,

I installed this plugin and getting some issues. When I process an order then its not waiting to get payment done and automatically completes the order. Order status automatically changed to processing.

It should wait for the successful payment and then order status change to successful.

Kindly check video
https://www.screencast.com/t/qR4C09dfjvZR

This is the website where I have installed this plugin.
https://dev.chapakazi.co.tz/shop

Would appreciate you help.

Kind regards,

Ahmed Asghar

**Comment by aasghar551** [comment]
Is there anyone who can answer?

**Comment by wallace-stev** [comment]
Hello @aasghar551 ,  just noticed the issue you raised with one of my client sites as well. I understand that it is an inconvenience you didn't expect. I'm consulting with a few WP developers and Selcom leads trying to solve this.

**Comment by aasghar551** [comment]
Hi @wallace-stev 

is there any update please?

**Comment by aasghar551** [comment]
Seems not taking care on the ticket.

**Comment by wallace-stev** [comment]
Hello @aasghar551 this will issue will be fixed in the next build approximately a week from now. I have revised the selcom documentation and found where the plugin fails to deliver. Please be patient while I'm working to resolve and run tests.

**Comment by aasghar551** [comment]
Will be waiting for next build.

**Comment by aasghar551** [comment]
Hi @wallace-stev

is there any update about new build?

**Comment by aasghar551** [comment]
Can you please help me to fix the issue? You said 1 week for next build but now over 1 month gone and didn't see the next build.

**Comment by wallace-stev** [comment]
Hello @aasghar551 kindly be informed that the issue has been addressed. You will notice that with the latest commit the order status is first updated to "Pending" and awaits for payment update. Then I have implemented an endpoint with a function to listen for incoming callbacks and complete the order by updating it's status. Furthermore a workaround to fetch order status and complete the order manually (with a click of a button) is being implemented.

**Comment by wallace-stev** [comment]
Hello @aasghar551 kindly be informed that the issue has been addressed. You will notice that with the latest commit the order status is first updated to "Pending" and awaits for payment update. Then I have implemented an endpoint with a function to listen for incoming callbacks and complete the order by updating it's status. Furthermore a workaround to fetch order status and complete the order manually (with a click of a button) is being implemented.
